src/main/groovy/org/shipkit/internal/gradle/plugin/tasks/PluginValidator.java

```diff
@@ -1,95 +1,129 @@
 package org.shipkit.internal.gradle.plugin.tasks;
 
 import org.gradle.api.GradleException;
+import org.gradle.api.Project;
 import org.gradle.api.logging.Logger;
 import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.SourceSet;
+import org.shipkit.internal.gradle.util.JavaPluginUtil;
+import org.shipkit.internal.gradle.util.StringUtil;
 
 import java.io.File;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
+import static org.shipkit.internal.gradle.plugin.tasks.PluginUtil.getImplementationClass;
 import static org.shipkit.internal.gradle.plugin.tasks.PluginUtil.DOT_PROPERTIES;
 
 
 public class PluginValidator {
 
     private static final Logger LOG = Logging.getLogger(PluginValidator.class);
+    private static final String[] PLUGIN_EXTENSIONS = new String[]{".groovy", ".java"};
 
-    public void validate(Set<File> gradlePlugins, Set<File> gradleProperties) {
-        ensurePluginsHavePropertiesFile(gradlePlugins, gradleProperties);
+    private Set<File> sourceDirs;
+
+    public void validate(Project project) {
+        SourceSet sourceSet = JavaPluginUtil.getMainSourceSet(project);
```

Comment:
In general, during the task execution, the task should not interrogate "project" object for the inputs. Instead, it's best if the task has a property that is set by the plugin. This way, the task is more robust, more useful, the user can reconfigure it easily.

You had it before - there was a "sourceSet" property on the task that you were setting in the plugin. That was really, good, any particular reason you removed it?